location/models.py

```diff
@@ -1,13 +1,41 @@
 from django.db import models
+from django.core.validators import RegexValidator
+
+valid_name = RegexValidator(r'^[&0-9a-zA-Z\s_\.\,-]+$',
+                            'Only alphanumeric characters, and _,.-_& are allowed.')
 
 
 class Location(models.Model):
-    name = models.CharField(max_length=40)
-    city = models.CharField(max_length=20)
-    street = models.CharField(max_length=20)
+    name = models.CharField(max_length=40, unique=True, validators=[valid_name])
+    city = models.CharField(max_length=20, validators=[valid_name])
+    street = models.CharField(max_length=20, validators=[valid_name])
     street_number = models.PositiveIntegerField()
     indoor = models.BooleanField()
     description = models.CharField(max_length=255)
 
     def __str__(self):
         return self.name
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=['city', 'street', 'street_number'],
+                                    name='unique address')
+        ]
+
+    def update_name(self, name):
+        self.name = name
+
+    def update_city(self, city):
+        self.city = city
+
+    def update_street(self, street):
+        self.street = street
+
+    def update_street_number(self, street_number):
+        self.street_number = street_number
+
+    def update_indoor(self, indoor):
+        self.indoor = indoor
+
+    def update_description(self, description):
+        self.description = description
```

Comment:
maybe you can do one update method with all the options like:
```suggestion
    def update(self, name=None, city=None, street=None, street_number=None, indoor=None, description=None):
        if name:
            self.name = name
        if city:
            self.city = city
        if street:
            self.street = street
        if street_number:
            self.street_number = street_number
        if indoor is not None:
            self.indoor = indoor
        if description:
            self.description = description
```

Reply:
now in the test file, you can do all the tests in one test.
also, I think you can do something like "test_update_of_invalid_values"